bin/propolis-server/src/lib/initializer.rs

```diff
@@ -842,6 +851,141 @@ impl<'a> MachineInitializer<'a> {
+        };
+
+        let vendor = cpuid::VendorKind::try_from(cpuid_vendor);
+        let proc_manufacturer = match vendor {
+            Ok(cpuid::VendorKind::Intel) => "Intel",
+            Ok(cpuid::VendorKind::Amd) => "Advanced Micro Devices, Inc.",
+            _ => "",
+        }
+        .try_into()
+        .unwrap();
+        let proc_family = match (vendor, family) {
+            // Explicitly match for Zen-based CPUs
+            //
+            // Although this family identifier is not valid in SMBIOS 2.7,
+            // having been defined in 3.x, we pass it through anyways.
+            (Ok(cpuid::VendorKind::Amd), family) if family >= 0x17 => 0x6b,
+
+            // Emit Unknown for everything else
+            _ => 0x2,
+        };
+        let proc_id =
+            u64::from(cpuid_ident.eax) | u64::from(cpuid_ident.edx) << 32;
+        let proc_version =
+            cpuid::parse_brand_string(cpuid_procname).unwrap_or("".to_string());
+
+        let smb_type4 = smbios::table::Type4 {
+            // central processor
+            proc_type: 0x03,
+            proc_family,
+            proc_manufacturer,
+            proc_id,
+            proc_version: proc_version.try_into().unwrap_or_default(),
+            // cpu enabled, socket populated
+            status: 0x41,
+            // unknown
+            proc_upgrade: 0x2,
+            // make core and thread counts equal for now
+            core_count: self.properties.vcpus,
+            core_enabled: self.properties.vcpus,
+            thread_count: self.properties.vcpus,
+            // 64-bit capable, multicore
+            proc_characteristics: 0xc,
+            ..Default::default()
+        };
+
+        let memsize_bytes = (self.properties.memory as usize) * MB;
+        let mut smb_type16 = smbios::table::Type16 {
+            // system board
+            location: 0x3,
+            // system memory
+            array_use: 0x3,
+            // unknown
+            error_correction: 0x2,
+            num_mem_devices: 1,
+            ..Default::default()
+        };
+        smb_type16.set_max_capacity(memsize_bytes);
+
+        let mut smb_type17 = smbios::table::Type17 {
+            // handle for type16 assigned below
+            phys_mem_array_handle: 0x1600.into(),
+            // Unknown
+            form_factor: 0x2,
+            // Unknown
+            memory_type: 0x2,
+            ..Default::default()
+        };
+        smb_type17.set_size(Some(memsize_bytes));
+
+        let smb_type32 = smbios::table::Type32::default();
+
+        let mut smb_tables = smbios::Tables::new(0x7f00.into());
+        smb_tables.add(0x0000.into(), &smb_type0).unwrap();
+        smb_tables.add(0x0100.into(), &smb_type1).unwrap();
+        smb_tables.add(0x0300.into(), &smb_type4).unwrap();
+        smb_tables.add(0x1600.into(), &smb_type16).unwrap();
+        smb_tables.add(0x1700.into(), &smb_type17).unwrap();
+        smb_tables.add(0x3200.into(), &smb_type32).unwrap();
```

Comment:
Per the SMBIOS specification (DSP0134 § 6.2, Table 4), the type 3 "system enclosure" table must have a non-null manufacturer string and type field. AFAICT, we're okay not providing a "cache information" (type 7) table as we're not pretending to have an off-die cache. I _think_ that "system information" (type 9) table is...maybe not necessary, as we're not pretending that the virtual hardware is upgradeable? OTTOH perhaps we ought to be populating it with our virtual PCI devices? I'm nottotally sure what we ought to do there...

Reply:
We are definitely not completely following the spec with the tables currently presented.  The tables listed is what @rmustacc and I thought would be appropriate to start out with.  We can definitely expand on it in the future, if required.

Reply:
Fair enough! A comment about what we do and don't populate and the justification for it might be nice to see if you have the time to add one, but I'm not going to block merging this over it.

Reply:
I wrote up a little comment there.